includes/compat/class-wc-stripe-subs-compat.php

```diff
@@ -267,7 +271,10 @@ public function process_subscription_payment( $amount = 0.0, $renewal_order, $re
 			$source_object   = $prepared_source->source_object;
 
 			if ( ! $prepared_source->customer ) {
-				return new WP_Error( 'stripe_error', __( 'Customer not found', 'woocommerce-gateway-stripe' ) );
+				throw new WC_Stripe_Exception(
+					'Customer not found while processing renewal for order ' . $renewal_order->get_id(),
```

Comment:
As we have both customer term in WooCommerce and Stripe it may be a bit confusing. At least it's sometimes confusing to me. How about making the message more explicit, e.g. `Failed to process renewal for order "orderId". Stripe customer id is missing in the order.`